src/Finder/CashOnDeliveryPaymentMethodFinder.php

```diff
@@ -11,17 +11,17 @@
 namespace BitBag\ShopwareOrlenPaczkaPlugin\Finder;
 
 use Shopware\Core\Checkout\Payment\Cart\PaymentHandler\CashPayment;
-use Shopware\Core\Checkout\Payment\DataAbstractionLayer\PaymentMethodRepositoryDecorator;
 use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
 use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
 use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
 use Shopware\Core\Framework\DataAbstractionLayer\Search\IdSearchResult;
 
 final class CashOnDeliveryPaymentMethodFinder implements CashOnDeliveryPaymentMethodFinderInterface
 {
-    private PaymentMethodRepositoryDecorator $paymentMethodRepository;
+    private EntityRepositoryInterface $paymentMethodRepository;
 
-    public function __construct(PaymentMethodRepositoryDecorator $paymentMethodRepository)
+    public function __construct(EntityRepositoryInterface $paymentMethodRepository)
```

Comment:
Can't we use EntityRepository instead? EntityRepositoryInterface is deprecated and will be removed in 6.5 version.